src/firebase-db.ts

```diff
@@ -409,3 +414,65 @@ export const setLearnerPluginState = async (pluginId: number, state: string): Pr
 
   return state;
 };
+
+export const getLegacyLinkedRefIds = (embeddableRefId: string, linkedRefMap: LegacyLinkedRefMap) => {
+  const linkedRefIds: string[] = [];
+  let refId: string | undefined = embeddableRefId;
+  do {
+    // break out if a cycle is found
+    if (linkedRefIds.indexOf(refId) !== -1) {
+      break;
+    }
+    if (refId !== embeddableRefId) {
+      linkedRefIds.push(refId);
+    }
+    refId = linkedRefMap[refId]?.linkedRefId;
+  } while (refId);
+
+  return linkedRefIds;
+};
+
+export const getLegacyLinkedInteractiveInfo = (embeddableRefId: string, laraData: ILaraData, callback: (info: ILegacyLinkedInteractiveState) => void) => {
+  // get a map of embeddable refs to linked refs
+  const linkedRefMap = getLegacyLinkedRefMap(laraData);
+
+  // if this ref isn't in the map it doesn't have linked interactives so we are done
+  if (!linkedRefMap[embeddableRefId]) {
+    callback({
+      hasLinkedInteractive: false,
+      linkedState: null,
+      allLinkedStates: []
+    });
+    return;
+  }
+
+  // get all the linked ref states in ancestry order with a guard against a loop in the graph
+  const linkedRefIds = getLegacyLinkedRefIds(embeddableRefId, linkedRefMap);
+
+  getAllAnswersInList(linkedRefIds)
+    .then(answers => {
+      const allLinkedStates = linkedRefIds.map((linkedRefId, index) => {
+        const linkedRef = linkedRefMap[linkedRefId];
+
+        /*
+          NOTE: Lara also returns the following which we don't have access to or don't make sense in AP
+          createdAt, updatedAt, interactiveStateUrl, interactive: {id, name}
+
+          TODO: add at least createdAt when implementing linked interactive UI
```

Comment:
I think we need updatedAt, so we can tell which answer has been modified most recently.
This might also have an impact on the interactive attachments api. I'd guess that when it modifies an attachment it isn't always changing the answer. So once there is an updatedAt field, the attachment api should cause this neew updatedAt field to get updated.

Reply:
Fixed in https://github.com/concord-consortium/activity-player/pull/273/commits/7e1f864831587dec8685f33d7cc7d97e921ae477